python_modules/libraries/dagstermill/dagstermill/manager.py

```diff
@@ -298,15 +303,20 @@ def yield_result(self, value, output_name="result"):
                 f"Expected one of {[str(output_def.name) for output_def in self.solid_def.output_defs]}"
             )
 
-        dagster_type = self.solid_def.output_def_named(output_name).dagster_type
-
-        # https://github.com/dagster-io/dagster/issues/2648
-        # dagstermill temporary file creation should use a more systematic and robust scheme
-        out_file = os.path.join(
-            self.marshal_dir, f"{self.context.solid_handle}-output-{output_name}"
+        # pass output value cross process boundary using io manager
+        step_context = self.context._step_context  # pylint: disable=protected-access
+        # Note: yield_result currently does not support DynamicOutput
+        step_output_handle = StepOutputHandle(
+            step_key=step_context.step.key, output_name=output_name
         )
+        output_context = step_context.get_output_context(step_output_handle)
+        io_manager = step_context.get_io_manager(step_output_handle)
+
+        # Note that we assume io manager is symmetric, i.e handle_input(handle_output(X)) == X
+        io_manager.handle_output(output_context, value)
```

Comment:
Was this not done already in the handle_output PR?

Reply:
ah updated - merged in master